Block/Catalog/Product/View.php

```diff
@@ -92,9 +92,11 @@ public function __construct(
      */
     private function getProduct()
     {
-        $this->product = $this->registry->registry('product');
-        if (!$this->product->getId()) {
-            throw new LocalizedException(__('Failed to initialize product'));
+        if(is_null($this->product)){
```

Comment:
pour éviter la complexité cyclomatique, il vaut mieux `return` asap `if (!is_null($this->product))` plutôt que d'imbriquer les `if { if { if {`

Reply:
Du coup plus besoin de tester le `getId()` ?